### PR TITLE
Added stubs for gradient descent and optimization algorithms for gradient descent

### PR DESCRIPTION
**Pull request by ysharc**
I think the guides are missing the important stuff of neural networks. So, added the stubs for gradient descent and optimization algorithms for gradient descent. Not sure, whether each of the optimization algorithms like Adam, Adagrad, rmsprop etc., should have their own sub-page or they go into the stub I just created.

### CODE REVIEW

**Comment by luandy64** [comment]
Hey @ysharc 

Since the Guides should serve as a introduction to the topics it contains, I think it's safe to just have introductions to Gradient Descent and Optimizations for Gradient Descent.

Then within the broad/general article, there can be links to other resources that explain more about particular algorithms.

Just my opinion, hopefully other people can chime in here

**Comment by luandy64** [comment]
Hey @ysharc 

Since the Guides should serve as a introduction to the topics it contains, I think it's safe to just have introductions to Gradient Descent and Optimizations for Gradient Descent.

Then within the broad/general article, there can be links to other resources that explain more about particular algorithms.

Just my opinion, hopefully other people can chime in here.
